use '= default' to define a trivial destructor

Needed for #10588

```
/home/runner/work/ddnet/ddnet/src/base/logger.h:73:10: error: use '= default' to define a trivial destructor [modernize-use-equals-default,-warnings-as-errors]
   73 |         virtual ~ILogger() {}
      |                 ^          ~~
      |                            = default;
```

Remove the lint or apply the fix?

--- a/src/engine/antibot.h
+++ b/src/engine/antibot.h
@@ -23,7 +23,6 @@ public:
 
 	// Commands
 	virtual void ConsoleCommand(const char *pCommand) = 0;
-
 	virtual ~IAntibot() = default;
 };
 

--- a/src/engine/antibot.h
+++ b/src/engine/antibot.h
@@ -24,7 +24,7 @@ public:
 	// Commands
 	virtual void ConsoleCommand(const char *pCommand) = 0;
 
-	virtual ~IAntibot(){};
+	virtual ~IAntibot() = default;
 };
 
 class IEngineAntibot : public IAntibot
@@ -41,7 +41,7 @@ public:
 	virtual bool OnEngineServerMessage(int ClientId, const void *pData, int Size, int Flags) = 0;
 	virtual bool OnEngineSimulateClientMessage(int *pClientId, void *pBuffer, int BufferSize, int *pOutSize, int *pFlags) = 0;
 
-	virtual ~IEngineAntibot(){};
+	virtual ~IEngineAntibot() = default;
 };
 
 #endif //ENGINE_ANTIBOT_H

--- a/src/engine/client/input.h
+++ b/src/engine/client/input.h
@@ -37,7 +37,7 @@ public:
 		CInput *Input() { return m_pInput; }
 
 	public:
-		CJoystick() {}
+		CJoystick() = default;
 		CJoystick(CInput *pInput, int Index, SDL_Joystick *pDelegate);
 		virtual ~CJoystick() = default;
 

--- a/src/engine/client/input.h
+++ b/src/engine/client/input.h
@@ -37,7 +37,6 @@ public:
 		CInput *Input() { return m_pInput; }
 
 	public:
-		CJoystick() = default;
 		CJoystick(CInput *pInput, int Index, SDL_Joystick *pDelegate);
 		virtual ~CJoystick() = default;
 

--- a/src/engine/client/serverbrowser_http.h
+++ b/src/engine/client/serverbrowser_http.h
@@ -10,7 +10,7 @@ class IHttp;
 class IServerBrowserHttp
 {
 public:
-	virtual ~IServerBrowserHttp() {}
+	virtual ~IServerBrowserHttp() = default;
 
 	virtual void Update() = 0;
 

--- a/src/engine/client/serverbrowser_ping_cache.h
+++ b/src/engine/client/serverbrowser_ping_cache.h
@@ -8,7 +8,7 @@ class IStorage;
 class IServerBrowserPingCache
 {
 public:
-	virtual ~IServerBrowserPingCache() {}
+	virtual ~IServerBrowserPingCache() = default;
 
 	virtual void Load() = 0;
 

--- a/src/engine/console.h
+++ b/src/engine/console.h
@@ -52,7 +52,6 @@ public:
 		IResult(int ClientId) :
 			m_NumArgs(0),
 			m_ClientId(ClientId) {}
-		IResult(const IResult &Other) = default;
 		virtual ~IResult() = default;
 
 		virtual int GetInteger(unsigned Index) const = 0;

--- a/src/engine/console.h
+++ b/src/engine/console.h
@@ -52,10 +52,8 @@ public:
 		IResult(int ClientId) :
 			m_NumArgs(0),
 			m_ClientId(ClientId) {}
-		IResult(const IResult &Other) :
-			m_NumArgs(Other.m_NumArgs),
-			m_ClientId(Other.m_ClientId) {}
-		virtual ~IResult() {}
+		IResult(const IResult &Other) = default;
+		virtual ~IResult() = default;
 
 		virtual int GetInteger(unsigned Index) const = 0;
 		virtual float GetFloat(unsigned Index) const = 0;
@@ -79,7 +77,7 @@ public:
 
 	public:
 		CCommandInfo() { m_AccessLevel = ACCESS_LEVEL_ADMIN; }
-		virtual ~CCommandInfo() {}
+		virtual ~CCommandInfo() = default;
 		const char *m_pName;
 		const char *m_pHelp;
 		const char *m_pParams;

--- a/src/engine/demo.h
+++ b/src/engine/demo.h
@@ -82,7 +82,7 @@ public:
 		TICK_NEXT, // go to the next tick
 	};
 
-	virtual ~IDemoPlayer() {}
+	virtual ~IDemoPlayer() = default;
 	virtual void SetSpeed(float Speed) = 0;
 	virtual void SetSpeedIndex(int SpeedIndex) = 0;
 	virtual void AdjustSpeedIndex(int Offset) = 0;
@@ -109,7 +109,7 @@ public:
 		REMOVE_FILE,
 	};
 
-	virtual ~IDemoRecorder() {}
+	virtual ~IDemoRecorder() = default;
 	virtual bool IsRecording() const = 0;
 	virtual int Stop(IDemoRecorder::EStopMode Mode, const char *pTargetFilename = "") = 0;
 	virtual int Length() const = 0;

--- a/src/engine/editor.h
+++ b/src/engine/editor.h
@@ -8,7 +8,7 @@ class IEditor : public IInterface
 {
 	MACRO_INTERFACE("editor")
 public:
-	virtual ~IEditor() {}
+	virtual ~IEditor() = default;
 	virtual void Init() = 0;
 	virtual void OnUpdate() = 0;
 	virtual void OnRender() = 0;

--- a/src/engine/favorites.h
+++ b/src/engine/favorites.h
@@ -26,7 +26,7 @@ public:
 		bool m_AllowPing;
 	};
 
-	virtual ~IFavorites() {}
+	virtual ~IFavorites() = default;
 
 	virtual TRISTATE IsFavorite(const NETADDR *pAddrs, int NumAddrs) const = 0;
 	// Only considers the addresses that are actually favorites.

--- a/src/engine/ghost.h
+++ b/src/engine/ghost.h
@@ -19,7 +19,7 @@ class IGhostRecorder : public IInterface
 {
 	MACRO_INTERFACE("ghostrecorder")
 public:
-	virtual ~IGhostRecorder() {}
+	virtual ~IGhostRecorder() = default;
 
 	virtual int Start(const char *pFilename, const char *pMap, const SHA256_DIGEST &MapSha256, const char *pName) = 0;
 	virtual void Stop(int Ticks, int Time) = 0;
@@ -32,7 +32,7 @@ class IGhostLoader : public IInterface
 {
 	MACRO_INTERFACE("ghostloader")
 public:
-	virtual ~IGhostLoader() {}
+	virtual ~IGhostLoader() = default;
 
 	virtual bool Load(const char *pFilename, const char *pMap, const SHA256_DIGEST &MapSha256, unsigned MapCrc) = 0;
 	virtual void Close() = 0;

--- a/src/engine/graphics.h
+++ b/src/engine/graphics.h
@@ -339,7 +339,7 @@ public:
 	{
 	public:
 		float m_X0, m_Y0, m_X1, m_Y1;
-		CLineItem() {}
+		CLineItem() = default;
 		CLineItem(float x0, float y0, float x1, float y1) :
 			m_X0(x0), m_Y0(y0), m_X1(x1), m_Y1(y1) {}
 	};
@@ -372,7 +372,7 @@ public:
 	struct CFreeformItem
 	{
 		float m_X0, m_Y0, m_X1, m_Y1, m_X2, m_Y2, m_X3, m_Y3;
-		CFreeformItem() {}
+		CFreeformItem() = default;
 		CFreeformItem(float x0, float y0, float x1, float y1, float x2, float y2, float x3, float y3) :
 			m_X0(x0), m_Y0(y0), m_X1(x1), m_Y1(y1), m_X2(x2), m_Y2(y2), m_X3(x3), m_Y3(y3) {}
 	};
@@ -380,7 +380,7 @@ public:
 	struct CQuadItem
 	{
 		float m_X, m_Y, m_Width, m_Height;
-		CQuadItem() {}
+		CQuadItem() = default;
 		CQuadItem(float x, float y, float w, float h) :
 			m_X(x), m_Y(y), m_Width(w), m_Height(h) {}
 	};
@@ -439,7 +439,7 @@ public:
 	{
 		int m_Index;
 		float m_R, m_G, m_B, m_A;
-		CColorVertex() {}
+		CColorVertex() = default;
 		CColorVertex(int i, float r, float g, float b, float a) :
 			m_Index(i), m_R(r), m_G(g), m_B(b), m_A(a) {}
 		CColorVertex(int i, ColorRGBA Color) :

--- a/src/engine/kernel.h
+++ b/src/engine/kernel.h
@@ -19,7 +19,7 @@ public:
 	IInterface() :
 		m_pKernel(nullptr) {}
 	virtual void Shutdown() {}
-	virtual ~IInterface() {}
+	virtual ~IInterface() = default;
 };
 
 #define MACRO_INTERFACE(Name) \
@@ -40,7 +40,7 @@ class IKernel
 public:
 	static IKernel *Create();
 	virtual void Shutdown() = 0;
-	virtual ~IKernel() {}
+	virtual ~IKernel() = default;
 
 	// templated access to handle pointer conversions and interface names
 	template<class TINTERFACE>

--- a/src/engine/server/databases/connection.h
+++ b/src/engine/server/databases/connection.h
@@ -19,7 +19,7 @@ class IDbConnection
 {
 public:
 	IDbConnection(const char *pPrefix);
-	virtual ~IDbConnection() {}
+	virtual ~IDbConnection() = default;
 	IDbConnection &operator=(const IDbConnection &) = delete;
 	virtual void Print(IConsole *pConsole, const char *pMode) = 0;
 

--- a/src/engine/server/register.h
+++ b/src/engine/server/register.h
@@ -10,7 +10,7 @@ struct CNetChunk;
 class IRegister
 {
 public:
-	virtual ~IRegister() {}
+	virtual ~IRegister() = default;
 
 	virtual void Update() = 0;
 	// Call `OnConfigChange` if you change relevant config variables

--- a/src/engine/shared/demo.h
+++ b/src/engine/shared/demo.h
@@ -42,7 +42,7 @@ class CDemoRecorder : public IDemoRecorder
 
 public:
 	CDemoRecorder(class CSnapshotDelta *pSnapshotDelta, bool NoMapData = false);
-	CDemoRecorder() {}
+	CDemoRecorder() = default;
 	~CDemoRecorder() override;
 
 	int Start(class IStorage *pStorage, class IConsole *pConsole, const char *pFilename, const char *pNetversion, const char *pMap, const SHA256_DIGEST &Sha256, unsigned MapCrc, const char *pType, unsigned MapSize, unsigned char *pMapData, IOHANDLE MapFile, DEMOFUNC_FILTER pfnFilter, void *pUser);
@@ -66,7 +66,7 @@ public:
 	class IListener
 	{
 	public:
-		virtual ~IListener() {}
+		virtual ~IListener() = default;
 		virtual void OnDemoPlayerSnapshot(void *pData, int Size) = 0;
 		virtual void OnDemoPlayerMessage(void *pData, int Size) = 0;
 	};

--- a/src/engine/shared/netban.h
+++ b/src/engine/shared/netban.h
@@ -173,7 +173,7 @@ public:
 	class IConsole *Console() const { return m_pConsole; }
 	class IStorage *Storage() const { return m_pStorage; }
 
-	virtual ~CNetBan() {}
+	virtual ~CNetBan() = default;
 	void Init(class IConsole *pConsole, class IStorage *pStorage);
 	void Update();
 

--- a/src/engine/shared/video.h
+++ b/src/engine/shared/video.h
@@ -10,7 +10,7 @@ typedef std::function<void(short *pFinalOut, unsigned Frames)> ISoundMixFunc;
 class IVideo
 {
 public:
-	virtual ~IVideo(){};
+	virtual ~IVideo() = default;
 
 	virtual bool Start() = 0;
 	virtual void Stop() = 0;

--- a/src/game/client/component.h
+++ b/src/game/client/component.h
@@ -161,9 +161,7 @@ public:
 	/**
 	 * The component virtual destructor.
 	 */
-	virtual ~CComponent()
-	{
-	}
+	virtual ~CComponent() = default;
 
 	/**
 	 * Gets the size of the non-abstract component.

--- a/src/game/client/components/menu_background.h
+++ b/src/game/client/components/menu_background.h
@@ -18,7 +18,7 @@ class CMenuMap : public CBackgroundEngineMap
 class CTheme
 {
 public:
-	CTheme() {}
+	CTheme() = default;
 	CTheme(const char *pName, bool HasDay, bool HasNight) :
 		m_Name(pName), m_HasDay(HasDay), m_HasNight(HasNight) {}
 
@@ -100,7 +100,7 @@ private:
 
 public:
 	CMenuBackground();
-	~CMenuBackground() override {}
+	~CMenuBackground() override = default;
 	int Sizeof() const override { return sizeof(*this); }
 
 	void OnInterfacesInit(CGameClient *pClient) override;

--- a/src/game/client/components/menu_background.h
+++ b/src/game/client/components/menu_background.h
@@ -18,7 +18,6 @@ class CMenuMap : public CBackgroundEngineMap
 class CTheme
 {
 public:
-	CTheme() = default;
 	CTheme(const char *pName, bool HasDay, bool HasNight) :
 		m_Name(pName), m_HasDay(HasDay), m_HasNight(HasNight) {}
 
@@ -100,7 +99,6 @@ private:
 
 public:
 	CMenuBackground();
-	~CMenuBackground() override = default;
 	int Sizeof() const override { return sizeof(*this); }
 
 	void OnInterfacesInit(CGameClient *pClient) override;

--- a/src/game/client/components/menus.h
+++ b/src/game/client/components/menus.h
@@ -373,7 +373,7 @@ protected:
 		int m_CustomSkinColorFeet;
 
 	public:
-		CFriendItem() {}
+		CFriendItem() = default;
 		CFriendItem(const CFriendInfo *pFriendInfo) :
 			m_pServerInfo(nullptr),
 			m_IsPlayer(false),

--- a/src/game/client/components/menus.h
+++ b/src/game/client/components/menus.h
@@ -373,7 +373,6 @@ protected:
 		int m_CustomSkinColorFeet;
 
 	public:
-		CFriendItem() = default;
 		CFriendItem(const CFriendInfo *pFriendInfo) :
 			m_pServerInfo(nullptr),
 			m_IsPlayer(false),

--- a/src/game/editor/mapitems/layer.h
+++ b/src/game/editor/mapitems/layer.h
@@ -39,9 +39,7 @@ public:
 		m_Readonly = false;
 	}
 
-	virtual ~CLayer()
-	{
-	}
+	virtual ~CLayer() = default;
 
 	virtual void BrushSelecting(CUIRect Rect) {}
 	virtual int BrushGrab(std::shared_ptr<CLayerGroup> pBrush, CUIRect Rect) { return 0; }

--- a/src/game/localization.h
+++ b/src/game/localization.h
@@ -32,7 +32,7 @@ class CLocalizationDatabase
 		unsigned m_ContextHash;
 		const char *m_pReplacement;
 
-		CString() {}
+		CString() = default;
 		CString(unsigned Hash, unsigned ContextHash, const char *pReplacement) :
 			m_Hash(Hash), m_ContextHash(ContextHash), m_pReplacement(pReplacement)
 		{

--- a/src/game/server/score.h
+++ b/src/game/server/score.h
@@ -40,7 +40,7 @@ class CScore
 
 public:
 	CScore(CGameContext *pGameServer, CDbConnectionPool *pPool);
-	~CScore() {}
+	~CScore() = default;
 
 	CPlayerData *PlayerData(int Id) { return &m_aPlayerData[Id]; }
 

--- a/src/game/server/score.h
+++ b/src/game/server/score.h
@@ -40,7 +40,6 @@ class CScore
 
 public:
 	CScore(CGameContext *pGameServer, CDbConnectionPool *pPool);
-	~CScore() = default;
 
 	CPlayerData *PlayerData(int Id) { return &m_aPlayerData[Id]; }
 

--- a/src/game/server/scoreworker.h
+++ b/src/game/server/scoreworker.h
@@ -133,7 +133,7 @@ struct CSqlScoreData : ISqlData
 	{
 	}
 
-	virtual ~CSqlScoreData(){};
+	virtual ~CSqlScoreData() = default;
 
 	char m_aMap[MAX_MAP_LENGTH];
 	char m_aGameUuid[UUID_MAXSTRSIZE];
@@ -194,7 +194,7 @@ struct CSqlTeamSaveData : ISqlData
 		ISqlData(std::move(pResult))
 	{
 	}
-	virtual ~CSqlTeamSaveData(){};
+	virtual ~CSqlTeamSaveData() = default;
 
 	char m_aClientName[MAX_NAME_LENGTH];
 	char m_aMap[MAX_MAP_LENGTH];
@@ -209,7 +209,7 @@ struct CSqlTeamLoadRequest : ISqlData
 		ISqlData(std::move(pResult))
 	{
 	}
-	virtual ~CSqlTeamLoadRequest(){};
+	virtual ~CSqlTeamLoadRequest() = default;
 
 	char m_aCode[128];
 	char m_aMap[MAX_MAP_LENGTH];
@@ -227,7 +227,7 @@ public:
 	{
 		Reset();
 	}
-	~CPlayerData() {}
+	~CPlayerData() = default;
 
 	void Reset()
 	{

--- a/src/game/server/scoreworker.h
+++ b/src/game/server/scoreworker.h
@@ -133,8 +133,6 @@ struct CSqlScoreData : ISqlData
 	{
 	}
 
-	virtual ~CSqlScoreData() = default;
-
 	char m_aMap[MAX_MAP_LENGTH];
 	char m_aGameUuid[UUID_MAXSTRSIZE];
 	char m_aName[MAX_MAP_LENGTH];
@@ -194,7 +192,6 @@ struct CSqlTeamSaveData : ISqlData
 		ISqlData(std::move(pResult))
 	{
 	}
-	virtual ~CSqlTeamSaveData() = default;
 
 	char m_aClientName[MAX_NAME_LENGTH];
 	char m_aMap[MAX_MAP_LENGTH];
@@ -209,7 +206,6 @@ struct CSqlTeamLoadRequest : ISqlData
 		ISqlData(std::move(pResult))
 	{
 	}
-	virtual ~CSqlTeamLoadRequest() = default;
 
 	char m_aCode[128];
 	char m_aMap[MAX_MAP_LENGTH];
@@ -227,7 +223,6 @@ public:
 	{
 		Reset();
 	}
-	~CPlayerData() = default;
 
 	void Reset()
 	{

